op_crates/fetch/26_fetch.js

```diff
@@ -985,8 +985,9 @@
         this.headers = new Headers(input.headers);
         this.credentials = input.credentials;
         this._stream = input._stream;
-      } else if (typeof input === "string") {
-        this.url = input;
+      } else {
+        // TODO(nayeemrmn): Base from `--location` when implemented and set.
+        this.url = new URL(String(input)).href;
```

Comment:
Relative URLs, including the original repro `new Request(undefined).url`, are now correctly blocked by #7369.